nb2/bot/slack_bot.py

```diff
@@ -153,39 +163,55 @@ def help(self):
             ">`nb2 random quote` Digs up a random memory from a random person.\n"
             '>`nb2 remember (that|when) <person> said "<quote>"` Stores a new quote, to forever '
             "remain in the planes of Nostalgia.\n"
-            ">`nb2 remind (me|<person>) of <person>` Digs up a memorable quote from the past, and "
+            ">`nb2 remind (me|<person>+) of <person>` Digs up a memorable quote from the past, and "
             "remind the person."
         )
 
         blocks = [{"type": "section", "text": {"type": "mrkdwn", "text": msg}}]
 
         return self.Result(ok=True, message=blocks)
 
-    def remember(self, target_slack_user_id: str, quote: str):
+    def remember(self, message: str) -> Result:
         """
-        Add quote to a NB's memory of a Person with target_slack_user_id, if they exist.
+        Add quote to a NB's memory of a Person, if they exist.
         If they don't exist, look up their info and create a new Person in memory first.
 
         Args:
-            target_slack_user_id: string representing slack user id for Person to insert
-                                  quote into.
-            quote: string representing content of Quote to store.
+            message: A message that was sent to the bot.
 
         Returns:
             A Result namedtuple.
         """
-        if not get_person_by_slack_user_id(target_slack_user_id):
-            user_info = self.fetch_user_info(target_slack_user_id)
-            name = user_info.get("name")
-            first_name = name.split()[0]
+        matched = re.match(self.REMEMBER_REGEX, message, re.I)
 
-            create_person_dto = CreatePersonDTO(
-                slack_user_id=target_slack_user_id, first_name=first_name
-            )
+        if not matched:
+            return self.unknown_error
 
-            create_person(create_person_dto)
+        target_user_id = trim_mention(matched.group("user_id"))
+        quote = matched.group("quote")
+        person = get_person(target_user_id)
 
-        add_quote_dto = AddQuoteDTO(slack_user_id=target_slack_user_id, content=quote)
+        # The Person doesn't exist, so create a new Person from information fetched
+        # from Slack. If the user id could not be found in Slack, create a ghost Person.
+        if person is None:
+            try:
+                user_info = self.fetch_user_info(target_user_id)
+                real_name = user_info.get("real_name")
+                first_name, *last_name = real_name.split()
+                display_name = user_info.get("name")
```

Comment:
"name" seems to be based on what's in you email and doesn't seem to be editable by the person directly. Which is probably why it was used for the ghost_id originally

It would be nice if we could update person details. Then using "display_name" makes sense and also means that we should update all places where the bot displays the first name to instead display the display_name

Reply:
this looks like an outdated comment and has been resolved